code/modules/fluids/fluid_objects.dm

```diff
@@ -441,6 +441,16 @@ TYPEINFO(/obj/machinery/fluid_canister)
 			user.set_loc(target)
 			user.show_text("You climb [src].")
 
+	Click(location, control, params)
+		if (isobserver(usr))
+			return src.attack_hand(usr)
+		..()
+
+	attack_ai(mob/user)
+		if (!user.stat && !user.getStatusDuration("weakened") && BOUNDS_DIST(user, src) <= 0)
```

Comment:
```suggestion
		if (can_act(user) && in_interact_range(src, usr))
```

Reply:
This is the more general and preferred check. Let me know in Discord if there was a good reason to avoid it.

Reply:
I did copy it from the ladder AI interaction (and reversed it). No particular other reason.